### PR TITLE
[664] Refactor mapper flock

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

Move flock to main from Mapper struct.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by makr11st** [comment]
The change seems fine, but I believe there should be some tests failing due to the change? Maybe some integration tests?

File: crates/core/tedge_mapper/src/main.rs [hunk 1]
```diff
@@ -55,14 +58,27 @@ pub enum MapperName {
...
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
     let mapper = MapperOpt::from_args();
     tedge_utils::logging::initialise_tracing_subscriber(mapper.debug);
 
     let component = lookup_component(&mapper.name);
-
     let config = tedge_config()?;
+    // Run only one instance of a mapper
+    let _flock = check_another_instance_is_not_running(&mapper.name.to_string())?;
```
[the remark this answers is not in the document]
Comment (PradeepKiruvale) [reply crates/core/tedge_mapper/src/main.rs:80]: Fixed, using the full mapper name as in the mapper code.

**Comment by PradeepKiruvale** [comment]
> The change seems fine, but I believe there should be some tests failing due to the change? Maybe some integration tests?

Ran all the tests that can run with `cargo test` none of them failed.

**Review by makr11st — APPROVED** [review]

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**